File: provision-uptime-kuma-sqlite.py
#!/usr/bin/env python3
"""Provision Uptime Kuma monitors + Discord notification via direct SQLite insert."""
import sqlite3, json, os, datetime

DB = os.path.expanduser("~/uptime-kuma/data/kuma.db")
WEBHOOK = open(os.path.expanduser("~/.openclaw/secrets/discord_bobatrades_webhook")).read().strip()
USER_ID = 1
NOW = datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S")

MONITORS = [
    ("MC Health",            "http://localhost:3000/api/health",           '"ok":true'),
    ("MC Status (Alpaca)",   "http://localhost:3000/api/status",            '"equity"'),
    ("MC Market Data",       "http://localhost:3000/api/market",            '"timestamp"'),
    ("MC Live Signals",      "http://localhost:3000/api/live-signals",      '"lastCycleTime"'),
    ("MC Signals Latest",    "http://localhost:3000/api/signals/latest",    None),
    ("MC Pipeline Feed",     "http://localhost:3000/api/pipeline-feed",     '"events"'),
    ("MC Agents",            "http://localhost:3000/api/agents",            '"success"'),
    ("MC Flow Database",     "http://localhost:3000/api/flow-database",     None),
    ("MC Options Flow",      "http://localhost:3000/api/options-flow",      None),
    ("MC Portfolio",         "http://localhost:3000/api/portfolio",         None),
    ("MC Trades",            "http://localhost:3000/api/trades",            None),
    ("MC Alerts DB",         "http://localhost:3000/api/alerts-db",        None),
    ("MC Regime",            "http://localhost:3000/api/regime",            None),
    ("MC Macro Regime",      "http://localhost:3000/api/macro-regime",      None),
    ("MC Calendar",          "http://localhost:3000/api/calendar",          None),
    ("MC Market News",       "http://localhost:3000/api/market-news",       None),
    ("MC Live Strategy",     "http://localhost:3000/api/live-strategy",     None),
    ("MC Desk Cycle",        "http://localhost:3000/api/desk-cycle",        None),
    ("MC Analyst Signals",   "http://localhost:3000/api/analyst-signals",   None),
    ("MC Telegram Signals",  "http://localhost:3000/api/telegram-signals",  None),
    ("MC Chart Signals",     "http://localhost:3000/api/chart-signals",     None),
    ("MC Activity",          "http://localhost:3000/api/activity",          None),
    ("MC Trade Log",         "http://localhost:3000/api/trade-log",         None),
    ("MC DB Status",         "http://localhost:3000/api/db-status",         None),
    ("MC Dashboard",         "http://localhost:3000/",                      None),
]

con = sqlite3.connect(DB)
cur = con.cursor()

print(f"Inserting {len(MONITORS)} monitors...")
for name, url, keyword in MONITORS:
    mtype = "keyword" if keyword else "http"
    cur.execute("""
        INSERT OR IGNORE INTO monitor
          (name, active, user_id, interval, url, type, weight, maxretries,
           ignore_tls, upside_down, maxredirects, accepted_statuscodes_json,
           retry_interval, method, created_date, keyword)
        VALUES (?,1,?,60,?,?,2000,2,0,0,10,'[\"200-299\"]',60,'GET',?,?)
    """, (name, USER_ID, url, mtype, NOW, keyword))
    print(f"  ✓ {name}")

# Discord notification
print("\nInserting Discord notification...")
config = json.dumps({
    "type": "discord",
    "webhookURL": WEBHOOK,
    "discordUsername": "Mission Control",
    "discordPrefixMessage": "🔴 MC Alert:",
})
cur.execute("""
    INSERT OR IGNORE INTO notification (name, active, user_id, is_default, config)
    VALUES ('Discord MC', 1, ?, 1, ?)
""", (USER_ID, config))

notif_id = cur.execute(
    "SELECT id FROM notification WHERE name='Discord MC' AND user_id=?", (USER_ID,)
).fetchone()[0]
print(f"  ✓ Discord notification id={notif_id} (is_default=1 — auto-applies to all monitors)")

# Link notification to all monitors
monitor_ids = [r[0] for r in cur.execute("SELECT id FROM monitor WHERE user_id=?", (USER_ID,)).fetchall()]
for mid in monitor_ids:
    cur.execute("INSERT OR IGNORE INTO monitor_notification (monitor_id, notification_id) VALUES (?,?)", (mid, notif_id))
print(f"  ✓ Linked Discord to {len(monitor_ids)} monitors")

con.commit()
con.close()
print("\nDone.")

File: provision-uptime-kuma-sqlite.js
#!/usr/bin/env node
/**
 * provision-uptime-kuma-sqlite.js
 * Inserts MC monitors + Discord notification directly into Uptime Kuma's SQLite DB
 * using better-sqlite3 (already installed by Uptime Kuma).
 */
'use strict';

const Database = require('/home/itsju/uptime-kuma/node_modules/better-sqlite3');
const fs = require('fs');
const path = require('path');

const DB_PATH = path.join(process.env.HOME, 'uptime-kuma/data/kuma.db');
const USER_ID = 1;
const DISCORD_WEBHOOK = fs.readFileSync(
  path.join(process.env.HOME, '.openclaw/secrets/discord_bobatrades_webhook'), 'utf-8'
).trim();

const db = new Database(DB_PATH);
const now = new Date().toISOString().replace('T', ' ').slice(0, 19);

const insertMonitor = db.prepare(`
  INSERT OR IGNORE INTO monitor
    (name, active, user_id, interval, url, type, weight, maxretries,
     ignore_tls, upside_down, maxredirects, accepted_statuscodes_json,
     retry_interval, method, created_date, keyword)
  VALUES
    (@name, 1, @user_id, 60, @url, @type, 2000, 2,
     0, 0, 10, @statuses,
     60, 'GET', @now, @keyword)
`);

const MONITORS = [
  { name: 'MC Health',           url: 'http://localhost:3000/api/health',           keyword: '"ok":true' },
  { name: 'MC Status (Alpaca)',  url: 'http://localhost:3000/api/status',            keyword: '"equity"' },
  { name: 'MC Market Data',     url: 'http://localhost:3000/api/market',            keyword: '"timestamp"' },
  { name: 'MC Live Signals',    url: 'http://localhost:3000/api/live-signals',      keyword: '"lastCycleTime"' },
  { name: 'MC Signals Latest',  url: 'http://localhost:3000/api/signals/latest',   keyword: null },
  { name: 'MC Pipeline Feed',   url: 'http://localhost:3000/api/pipeline-feed',    keyword: '"events"' },
  { name: 'MC Agents',          url: 'http://localhost:3000/api/agents',            keyword: '"success"' },
  { name: 'MC Flow Database',   url: 'http://localhost:3000/api/flow-database',    keyword: null },
  { name: 'MC Options Flow',    url: 'http://localhost:3000/api/options-flow',     keyword: null },
  { name: 'MC Portfolio',       url: 'http://localhost:3000/api/portfolio',        keyword: null },
  { name: 'MC Trades',          url: 'http://localhost:3000/api/trades',           keyword: null },
  { name: 'MC Alerts DB',       url: 'http://localhost:3000/api/alerts-db',        keyword: null },
  { name: 'MC Regime',          url: 'http://localhost:3000/api/regime',           keyword: null },
  { name: 'MC Macro Regime',    url: 'http://localhost:3000/api/macro-regime',     keyword: null },
  { name: 'MC Calendar',        url: 'http://localhost:3000/api/calendar',         keyword: null },
  { name: 'MC Market News',     url: 'http://localhost:3000/api/market-news',      keyword: null },
  { name: 'MC Live Strategy',   url: 'http://localhost:3000/api/live-strategy',    keyword: null },
  { name: 'MC Desk Cycle',      url: 'http://localhost:3000/api/desk-cycle',       keyword: null },
  { name: 'MC Analyst Signals', url: 'http://localhost:3000/api/analyst-signals',  keyword: null },
  { name: 'MC Telegram Signals',url: 'http://localhost:3000/api/telegram-signals', keyword: null },
  { name: 'MC Chart Signals',   url: 'http://localhost:3000/api/chart-signals',    keyword: null },
  { name: 'MC Activity',        url: 'http://localhost:3000/api/activity',         keyword: null },
  { name: 'MC Trade Log',       url: 'http://localhost:3000/api/trade-log',        keyword: null },
  { name: 'MC DB Status',       url: 'http://localhost:3000/api/db-status',        keyword: null },
  { name: 'MC Dashboard',       url: 'http://localhost:3000/',                     keyword: null },
];

console.log(`Inserting ${MONITORS.length} monitors...`);
const insertMany = db.transaction(() => {
  for (const m of MONITORS) {
    insertMonitor.run({
      name: m.name,
      url: m.url,
      type: m.keyword ? 'keyword' : 'http',
      keyword: m.keyword,
      user_id: USER_ID,
      statuses: '["200-299"]',
      now,
    });
    console.log(`  ✓ ${m.name}`);
  }
});
insertMany();

// Insert Discord notification
console.log('\nInserting Discord notification...');
const discordConfig = JSON.stringify({
  type: 'discord',
  webhookURL: DISCORD_WEBHOOK,
  discordUsername: 'Mission Control',
  discordPrefixMessage: '🔴 MC Alert:',
});

db.prepare(`
  INSERT OR IGNORE INTO notification (name, active, user_id, is_default, config)
  VALUES ('Discord MC', 1, ?, 1, ?)
`).run(USER_ID, discordConfig);

const notifId = db.prepare(`SELECT id FROM notification WHERE name='Discord MC' AND user_id=?`).get(USER_ID).id;
console.log(`  ✓ Discord notification id=${notifId} (is_default=1)`);

// Link to all monitors
const monitorIds = db.prepare(`SELECT id FROM monitor WHERE user_id=?`).all(USER_ID).map(r => r.id);
const link = db.prepare(`INSERT OR IGNORE INTO monitor_notification (monitor_id, notification_id) VALUES (?, ?)`);
const linkAll = db.transaction(() => { for (const mid of monitorIds) link.run(mid, notifId); });
linkAll();
console.log(`  ✓ Linked Discord to ${monitorIds.length} monitors`);

db.close();
console.log('\nDone. Restarting Uptime Kuma...');
